rskj-core/src/main/java/org/ethereum/util/RLP.java

```diff
@@ -191,43 +195,55 @@ public static BigInteger decodeBigInteger(byte[] data, int index) {
         return BigIntegers.fromUnsignedByteArray(bytes);
     }
 
+    private static void assertLengthOfLength(byte[] payload,byte lengthOfLength) {
+        if (lengthOfLength >= payload.length) {
```

Comment:
it seems that `lengthOfLength` means a size of the length value in bytes, not a position inside the `payload` array. Perhaps it should be checked that `(pos + lengthOfLength)` is within boundaries. Plus it'd make sense to also check if `lengthOfLength` isn't negative?

Reply:
Addressed